src/Forte.Optimizely.ContentUsage/Api/Features/ContentUsage/ContentUsageController.cs

```diff
@@ -31,24 +31,45 @@ public ContentUsageController(IContentTypeRepository contentTypeRepository, Cont
     [SwaggerResponse(StatusCodes.Status200OK, null, typeof(IEnumerable<ContentUsageDto>))]
     [SwaggerResponse(StatusCodes.Status404NotFound)]
     [Route("[action]", Name = GetContentUsagesRouteName)]
-    public ActionResult GetContentUsages([FromQuery] [BindRequired] Guid guid)
+    public ActionResult GetContentUsages([FromQuery] GetContentUsagesQuery queryData)
     {
-        var contentType = _contentTypeRepository.Load(guid);
+        var contentType = _contentTypeRepository.Load(queryData.Guid);
 
         if (contentType == null) return NotFound();
 
-        var contentUsages = _contentUsageService.GetContentUsages(contentType);
+        var contentUsagesQuery = _contentUsageService.GetContentUsages(contentType);
 
-        var contentUsagesDto = contentUsages.Select(contentUsage => new ContentUsageDto
+        contentUsagesQuery = string.IsNullOrEmpty(queryData.Query)
+            ? contentUsagesQuery
+            : contentUsagesQuery.Where(x => x.Name.ToLowerInvariant().Contains(queryData.Query.ToLowerInvariant()));
+        
+        const int itemsPerPage = 25;
+        var contentUsagesDto = contentUsagesQuery
+            .Sort(queryData)
+            .Skip(queryData.Page - 1 * itemsPerPage)
```

Comment:
should we handle case in which Page is not passed in query? (default value of int which result in strange behaviour I guess)